.github/workflows/codeql-analysis.yml

```diff
@@ -32,6 +32,12 @@ jobs:
     - name: Checkout repository
       uses: actions/checkout@v2
 
+    - name: Dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get upgrade -y
```

Comment:
I wouldn't do the upgrade here, that could potentially upgrade the whole system including kernel etc